samcli/local/apigw/path_converter.py

```diff
@@ -4,16 +4,31 @@
 
 import re
 
-FLASK_PATH_PARAMS = "/<path:proxy>"
-APIGW_PATH_PARAMS_ESCAPED = r"/{proxy\+}"
-APIGW_PATH_PARAMS = "/{proxy+}"
+# The regex captures any path information before the {proxy+}. This is to support paths that have other params in
+# them. Example: /id/{id}/user/{proxy+}. Otherwise the regex will match the first { with the last +} giving incorrect
+# results.
+APIGW_PATH_PARAMS_ESCAPED = r"(.*/){(.*)\+}"
```

Comment:
This is not any path param, but only wildcard ones ie. the ones with + at the end. Right? 

Can you update the comment and variable name to be more descriptive?

Reply:
Updated name and added a full example.

The comment seems correct though. It is describing that it captures the path before the {proxy+}. Without this capture, the regex will match the first '{' and the last '+}', which is incorrect if there are path parameters in the full path. 

Is there a part of this that is confusing that could be cleared up further?

Reply:
> path information before the {proxy+}
The "before" word gave me the impression that it is used to get everything in the string before the substring {proxy+}. Given /id/{id}/user/{proxy+}, is this regex used to return `id` and `proxy`? or just `proxy`?